routes/project/billing.rb

```diff
@@ -132,22 +129,17 @@ class Clover
         end
 
         r.is String do |pm_ubid|
-          payment_method = PaymentMethod.from_ubid(pm_ubid)
-
-          unless payment_method
-            response.status = 404
-            r.halt
-          end
+          next unless (payment_method = PaymentMethod.from_ubid(pm_ubid))
 
           r.delete true do
             unless payment_method.billing_info.payment_methods.count > 1
               response.status = 400
-              return {message: "You can't delete the last payment method of a project."}
+              next {message: "You can't delete the last payment method of a project."}
             end
 
             payment_method.destroy
 
-            return {message: "Deleting #{payment_method.ubid}"}
+            {message: "Deleting #{payment_method.ubid}"}
```

Comment:
nit: Since it's a response to an AJAX request, a 204 No Content status is also acceptable

Reply:
I pushed a commit to return empty 204 responses for this and the other case you found.